[Headers][X86] Allow basic AVX512 predicate ops to be used in constexpr

fennecJ opened the pull request:
Fix #158646

fennecJ committed:
Allow kand* to be used in constexpr

fennecJ committed:
Allow int2mask, mask2int to be used in constexpr

RKSimon reviewed (commented):
Thanks @fennecJ this structure looks fine to me - I've added kadd to the intrinsic list in #158646 as it should be as easy to support as the logic ops now.

RKSimon reviewed (changes requested):
Add the other mask logic / kadd variants to the patch as well.

fennecJ committed:
Add constexpr testcase for _kand_mask16

_kand_mask16 is an alias of _mm512_kand.
Although they are semantically identical, the existing test suite
contains a dedicated set of tests for _kand_mask16.
This commit follows that convention by adding the same constexpr
test case for completeness.

fennecJ committed:
Allow kandn* to be used in constexpr

fennecJ committed:
Allow kor* to be used in constexpr

fennecJ committed:
Allow kxnor* to be used in constexpr

fennecJ committed:
Remove too verbose comments

Some comments in the constexpr test suite were a bit too verbose. This
commit removes them to keep the code concise.

fennecJ committed:
Allow kxor* to be used in constexpr

fennecJ committed:
Allow knot* to be used in constexpr

fennecJ committed:
Allow kadd* to be used in constexpr

RKSimon committed:
Merge branch 'main' into AVX512_Predicate_ConstExpr

RKSimon reviewed (approved):
LGTM - cheers!

RKSimon committed:
Merge branch 'main' into AVX512_Predicate_ConstExpr
